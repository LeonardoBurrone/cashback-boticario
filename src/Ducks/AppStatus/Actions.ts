// TODO: testar actions
import { AppStatusActionTypes, ChangeIsLoggedInAction, ChangeLoadingAction, ChangeLoadingMessageAction } from './types';

export const changeIsLoggedInAction = (payload: boolean): ChangeIsLoggedInAction => ({
  payload,
  type: AppStatusActionTypes.CHANGE_IS_LOGGED_IN
});

export const changeLoadingAction = (payload: boolean): ChangeLoadingAction => ({
  payload,
  type: AppStatusActionTypes.CHANGE_LOADING
});

export const changeLoadingMessageAction = (payload: string): ChangeLoadingMessageAction => ({
  payload,
  type: AppStatusActionTypes.CHANGE_LOADING_MESSAGE
});
