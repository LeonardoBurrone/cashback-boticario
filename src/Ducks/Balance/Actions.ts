import { BalanceActionTypes, ChangeBalanceAction, ChangeBalanceErrorAction, FetchBalanceAction } from './types';

export const changeBalanceAction = (payload: string): ChangeBalanceAction => ({
  payload,
  type: BalanceActionTypes.CHANGE_BALANCE
});

export const changeBalanceErrorAction = (payload: boolean): ChangeBalanceErrorAction => ({
  payload,
  type: BalanceActionTypes.CHANGE_BALANCE_ERROR
});

export const fetchBalanceAction = (): FetchBalanceAction => ({
  type: BalanceActionTypes.FETCH_BALANCE
});
