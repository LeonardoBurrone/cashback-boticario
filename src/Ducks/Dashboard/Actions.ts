import {
  ChangePuchasesAction,
  DashboardActionTypes,
  FetchPuschasesAction,
  FetchPuschasesErrorAction,
  Purchase
} from './types';

export const changePuchasesAction = (payload: Purchase[]): ChangePuchasesAction => ({
  payload,
  type: DashboardActionTypes.CHANGE_PURCHASES
});

export const fetchPurchasesAction = (): FetchPuschasesAction => ({
  type: DashboardActionTypes.FETCH_PURCHASES
});

export const fetchPurchasesErrorAction = (payload: boolean): FetchPuschasesErrorAction => ({
  payload,
  type: DashboardActionTypes.FETCH_PURCHASES_ERROR
});
