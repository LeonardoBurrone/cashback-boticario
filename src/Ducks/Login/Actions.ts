import { AuthenticateAction, LoginAction, LoginActionTypes, LogoutAction } from './types';

export const authenticateAction = (): AuthenticateAction => ({
  type: LoginActionTypes.AUTHENTICATE
});

export const loginAction = (payload: { email: string; password: string }): LoginAction => ({
  payload,
  type: LoginActionTypes.LOGIN
});

export const logoutAction = (): LogoutAction => ({
  type: LoginActionTypes.LOGOUT
});
