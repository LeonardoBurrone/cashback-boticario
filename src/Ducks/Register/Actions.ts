import { RegisterActionTypes, RegisterPurchaseAction } from './types';

export const registerPurchaseAction = (payload: {
  code: string;
  date: string;
  price: string;
}): RegisterPurchaseAction => ({
  payload,
  type: RegisterActionTypes.REGISTER_PURCHASE
});
