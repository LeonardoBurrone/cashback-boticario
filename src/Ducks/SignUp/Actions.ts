// TODO: testar actions
import { SignActionTypes, SignUpAction } from './types';

export const signUpAction = (payload: {
  document: string;
  email: string;
  name: string;
  password: string;
}): SignUpAction => ({
  payload,
  type: SignActionTypes.SIGN_UP
});
