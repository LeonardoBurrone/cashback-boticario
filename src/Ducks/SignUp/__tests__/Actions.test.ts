import configureMockStore from 'redux-mock-store';
import { signUpAction } from '../Actions';

describe('Test Sign Up Actions', () => {
  const mockStore = configureMockStore();
  const store = mockStore({});

  beforeEach(() => {
    store.clearActions();
  });

  it('Check if "signUpAction" was dispatched', () => {
    store.dispatch(
      signUpAction({
        document: '11122233345',
        email: 'test@test.com',
        name: 'Test Name',
        password: '123456'
      })
    );

    const actions = store.getActions();
    expect(actions).toMatchSnapshot();
  });
});
