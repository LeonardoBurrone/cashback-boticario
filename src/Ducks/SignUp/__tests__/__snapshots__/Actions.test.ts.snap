// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Sign Up Actions Check if "signUpAction" was dispatched 1`] = `
Array [
  Object {
    "payload": Object {
      "document": "11122233345",
      "email": "test@test.com",
      "name": "Test Name",
      "password": "123456",
    },
    "type": "SIGN_UP",
  },
]
`;
